PYIC-8254: Add EVCS PUT contract tests

MikeCollingwood opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

- Add put /vcs contract tests

### Why did it change

- Contract test coverage

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-8254](https://govukverify.atlassian.net/browse/PYIC-8254)

## Checklists

<!-- Delete if changes in READMEs or documentation are not required -->
- [x] All READMEs and documentation updated where necessary

<!-- Delete if changes don't include risk of credentials being exposed -->
- [x] No risk of PII, credentials or anything else sensitive being exposed through logs

<!-- Delete if changes don't apply -->
- [x] API/unit/contract tests have been written/updated

<!-- Delete if changes don't apply -->
- [x] Production changes appropriately staged out


[PYIC-8254]: https://govukverify.atlassian.net/browse/PYIC-8254?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

MikeCollingwood committed:
PYIC-8254: Add early changes for EVCS PUT contract testing

MikeCollingwood committed:
PYIC-8254: Add additional contract test cases

MikeCollingwood committed:
Merge branch 'main' into pyic-8254

MikeCollingwood committed:
PYIC-8254: SpotlessApply lint fix

MikeCollingwood committed:
PYIC-8254: Add extra tests and SpotlessApply lint fix

MikeCollingwood committed:
PYIC-8254: Add valid SI string

MikeCollingwood committed:
PYIC-8254: Add another 400 test

thebauSoftwire commented on libs/evcs-service/src/test/java/uk/gov/di/ipv/core/library/evcs/pact/ContractTest.java:
```diff
@@ -455,7 +516,383 @@ void testUpdateVcRequestReturns400(MockServer mockServer) {
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "putVcsAndStoredIdentity")
```
should we also have a test for an invalid SI object? E.g.a missing vot

thebauSoftwire commented on libs/evcs-service/src/test/java/uk/gov/di/ipv/core/library/evcs/pact/ContractTest.java:
```diff
@@ -455,7 +516,383 @@ void testUpdateVcRequestReturns400(MockServer mockServer) {
+    }
+
+    @Pact(provider = "EvcsProvider", consumer = "IpvCoreBack")
+    public RequestResponsePact putVcsWithoutApiKey(PactDslWithProvider builder) {
```
minor thing: should this be `putVcsWithInvalidApiKey`?

thebauSoftwire commented on libs/evcs-service/src/test/java/uk/gov/di/ipv/core/library/evcs/pact/ContractTest.java:
```diff
@@ -69,7 +72,65 @@ class ContractTest {
+                    List.of(
+                            new EvcsCreateUserVCsDto(
+                                    ADDRESS_VC_STRING, EvcsVCState.HISTORIC, Map.of(), ONLINE)),
+                    null);
```
could we update this so that we're passing in an updated address vc string and EvcsVCState.CURRENT to more closely match how we use the put method currently? With put, EVCS handles the updating of the states for existing VCs

MikeCollingwood replied:
Ah good to know thank you, I haven't been that close to it so ty will do

MikeCollingwood committed:
PYIC-8254: PR markups:
- Rename contract test to ref invalid api key
- Add invalid SI test
- Use updated address VC for realism in test

MikeCollingwood committed:
PYIC-8254: Add high-entropy strings to pragma allow-list

thebauSoftwire reviewed (approved):
LGTM!
